lib/adapters/xhr.js

```diff
@@ -82,7 +82,10 @@ module.exports = function xhrAdapter(resolve, reject, config) {
   request.onerror = function handleError() {
     // Real errors are hidden from us by the browser
     // onerror should only fire if it's a network error
-    reject(new Error('Network Error'));
+    var err = new Error('Network Error');
+    err.config = config;
```

Comment:
Added it! I understand and agree that the error handling should be improved, but in the meantime, it would be awesome to have this merged. I guess the output would be a lot like this in the final implementation...